### PR TITLE
DOCSP-37847: Add GraalVM to what's new

### PR DESCRIPTION
**Pull request by norareidy**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-java/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-37847
Staging - https://preview-mongodbnorareidy.gatsbyjs.io/java/DOCSP-37847-graalvm-info/whats-new/

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [ ] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

**Review by jordan-smith721 — APPROVED** [review]
LGTM

File: source/whats-new.txt [hunk 1]
```diff
@@ -35,6 +36,22 @@ Learn what's new in:
...
+-----------------
+
+.. _java-new-features-5.1:
+
+New Features in 5.1
+~~~~~~~~~~~~~~~~~~~
+
+The 5.1 driver release includes the following new features:
+
+- Support for building native applications by using the GraalVM native-image tool. These
```
Comment (vbabanin) [inline source/whats-new.txt:52]: We have begun testing with GraalVM native image technology, building and running a native application using the GraalVM native-image tool in our CI. However, we have not introduced specific support for end-users in `5.1`; I don't think it should be considered as a feature. Creating a native-image had been possible before 5.1, we only started actively testing this capability in version 5.1.

File: source/whats-new.txt [hunk 2]
```diff
@@ -35,6 +36,22 @@ Learn what's new in:
...
+  applications can access MongoDB through the driver's synchronous and reactive Java APIs.
```
Comment (vbabanin) [inline source/whats-new.txt:53]: We have begun testing with GraalVM native image technology, building and running a native application using the GraalVM native-image tool in our CI. However, we have not introduced specific support for end-users in `5.1`; I don't think it should be considered as a feature. Creating a native-image had been possible before 5.1, we only started actively testing this capability in version `5.1.`

**Review by vbabanin — CHANGES_REQUESTED** [review]

File: source/whats-new.txt (hunk 2 above)
Comment (norareidy) [reply source/whats-new.txt]: Thanks for the clarification - let me know if the updated wording works better!

**Review by vbabanin — APPROVED** [review]

**Pull request merged**